core/warning/warn_spec.rb

```diff
@@ -0,0 +1,27 @@
+require File.expand_path("../../../spec_helper", __FILE__)
+
+describe "Warning.warn" do
+  ruby_version_is "2.4" do
+    it "complains" do
+      -> {
+        Warning.warn("Chunky bacon!")
+      }.should complain("Chunky bacon!")
+    end
+
+    it "extends itself" do
+      Warning.ancestors.should include(Warning)
```

Comment:
This is true of any Module/Class.
I think we want: ` Warning.singleton_class.ancestors.should include(Warning)`